#110 Table 34 Patients With Serious Adverse Events by System Organ Class, FMQ and Preferred Term

yichen-wang1 opened the pull request:
closes #110

edelarua commented on R/fda-table_34.R:
```diff
@@ -0,0 +1,74 @@
+#' @inheritParams argument_convention
+#'
+#' @examples
+#' adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae")
+#' adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl")
+#'
+#' adae <- dplyr::rename(adae, FMQ01SC = SMQ01SC, FMQ01NAM = SMQ01NAM)
+#' levels(adae$FMQ01SC) <- c("BROAD", "NARROW")
+#' adae$FMQ01SC[is.na(adae$FMQ01SC)] <- "NARROW"
+#'
+#' tbl <- make_table_34(adae, adsl)
+#' tbl
+#'
```
Dataset can be more randomized to improve the output of the example.

```suggestion
#' library(dplyr)
#'
#' adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae")
#' adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl")
#'
#' set.seed(1)
#' adae <- adae %>%
#'   dplyr::rename(FMQ01SC = SMQ01SC) %>%
#'   dplyr::mutate(
#'     AESER = sample(c("Y", "N"), size = nrow(adae), replace = TRUE),
#'     FMQ01NAM = sample(c("FMQ1", "FMQ2", "FMQ3"), size = nrow(adae), replace = TRUE)
#'   )
#' adae$FMQ01SC[is.na(adae$FMQ01SC)] <- "NARROW"
#'
#' tbl <- make_table_34(adae = adae, alt_counts_df = adsl)
#' tbl
#'
```

edelarua commented on R/fda-table_34.R:
```diff
@@ -0,0 +1,74 @@
+#' tbl
+#'
+#' @export
+
```
Remove unnecessary whitespace
```suggestion
```

edelarua commented on R/fda-table_34.R:
```diff
@@ -0,0 +1,74 @@
+                          pref_var = "AETERM",
+                          lbl_pref_var = formatters::var_labels(adae, fill = TRUE)[pref_var],
+                          lbl_overall = NULL,
+                          prune_0 = FALSE,
```
Change default pruning setting to `TRUE`
```suggestion
                          prune_0 = TRUE,
```

edelarua commented on R/fda-table_34.R:
```diff
@@ -0,0 +1,74 @@
+                          show_colcounts = TRUE,
+                          arm_var = "ARM",
+                          fmqsc_var = "FMQO1SC",
+                          fmqnam_var = "FMQ01NAM",
```
We can include a `fmq_scope` variable to generalize in case users want to use "Broad" instead of "Narrow" scope
```suggestion
                          fmqnam_var = "FMQ01NAM",
                          fmq_scope = "NARROW",
```

edelarua commented on R/fda-table_34.R:
```diff
@@ -0,0 +1,74 @@
+                          alt_counts_df = NULL,
+                          show_colcounts = TRUE,
+                          arm_var = "ARM",
+                          fmqsc_var = "FMQO1SC",
```
Typo
```suggestion
                          fmqsc_var = "FMQ01SC",
```

edelarua commented on R/fda-table_34.R:
```diff
@@ -0,0 +1,74 @@
+                          annotations = NULL) {
+  checkmate::assert_subset(c("SAFFL", "USUBJID", "AEBODSYS", arm_var, fmqnam_var, fmqnam_var, pref_var), names(adae))
+  assert_flag_variables(adae, "SAFFL")
+
+  adae <- adae %>%
+    filter(SAFFL == "Y") %>%
+    filter(AESER == "Y") %>%
+    df_explicit_na(na_level = na_level)
+
+  adae <- adae[adae[[fmqsc_var]] == "NARROW", ]
+  adae[[fmqnam_var]] <- with_label(adae[[fmqnam_var]], "FMQ (Narrow)")
```
Implement the `fmq_scope` variable (see [table 10](https://github.com/pharmaverse/falcon/blob/main/R/fda-table_10.R) which also has a `fmq_scope` parameter)
```suggestion
  checkmate::assert_subset(toupper(fmq_scope), c("NARROW", "BROAD"))

    adae <- adae %>%
      filter(SAFFL == "Y", AESER == "Y", adae[[fmqsc_var]] == fmq_scope) %>%
      df_explicit_na(na_level = na_level)

    adae[[fmqnam_var]] <- with_label(adae[[fmqnam_var]], paste0("FMQ (", tools::toTitleCase(tolower(fmq_scope)), ")"))

```

edelarua reviewed (changes requested):
Hi @yichen-wang1,

Thanks for working on this! It looks great so far, just a few suggestions from me below - mainly an improvement for the example and implementing the `fmq_scope` variable to allow for more customization by the user. 

After updating these please also update documentation/tests (if needed) and then also create a website template for the table. You can use the other templates in the quarto/table-templates folder as a guide for your template and [this comment](https://github.com/pharmaverse/falcon/issues/64#issue-1746643359) for more details on what to update when adding a template.

Please let me know if you have any questions about this!! Thanks :)

yichen-wang1 commented:
> Hi @yichen-wang1,
> 
> Thanks for working on this! It looks great so far, just a few suggestions from me below - mainly an improvement for the example and implementing the `fmq_scope` variable to allow for more customization by the user.
> 
> After updating these please also update documentation/tests (if needed) and then also create a website template for the table. You can use the other templates in the quarto/table-templates folder as a guide for your template and [this comment](https://github.com/pharmaverse/falcon/issues/64#issue-1746643359) for more details on what to update when adding a template.
> 
> Please let me know if you have any questions about this!! Thanks :)

Hi @edelarua 

Thanks for the review. I updated the function and added the fmq_scope. I also created the website template according to other template files. Please let me know if anything is missing. Thank you! :)

edelarua reviewed (approved):
Hi @yichen-wang1,

It looks great! Thanks for adding the template in. I just reran tests/roxygen to update the documentation, made a few minor adjustments, and updated the NEWS file. I also fixed a bunch of checks that were failing but they were all unrelated to your table :)

Feel free to merge!
